tests/tom-swifty-test/SwiftReflector/ExpermimentalTests.cs

```diff
@@ -0,0 +1,255 @@
+﻿using System;
+using System.Runtime.InteropServices;
+using Dynamo.CSLang;
+using NUnit.Framework;
+using SwiftRuntimeLibrary;
+using SwiftRuntimeLibrary.SwiftMarshal;
+using tomwiftytest;
+
+namespace SwiftReflector
+{
+	[TestFixture]
+	public class ExpermimentalTests
+	{
+		[Test]
+		public void SingleIntStruct ()
+		{
+			var swiftCode = @"
+@frozen
+public struct OneInt {
+	public var X:Int
+	public init () { X = 17 }
+}
+";
+
+			// C# aux code:
+			// public struct XXOpaque1 {
+			// nint x0
+			// public static unsafe nint Getter (OneInt self)
+			// {
+			//    fixed (byte* thisSwiftDataPtr = SwiftRuntimeLibrary.SwiftMarshal.StructMarshal.Marshaler.PrepareValueType (self)) {
+			//        var p = (XXOpaque1 *)thisSwiftDataPtr;
+			//        return XGetter (*p);
+			// }
+			// [DllImport ("libExpermimentalTests.dylib", EntryPoint = "_$s18ExpermimentalTests6OneIntV1XSivg")]
+			// public static extern nint XGetter (XXOpaque1 r);
```

Comment:
Declaring the argument as a `byte*` makes the code simpler:

```suggestion
			// public static unsafe nint Getter (OneInt self)
			// {
			//    fixed (byte* thisSwiftDataPtr = SwiftRuntimeLibrary.SwiftMarshal.StructMarshal.Marshaler.PrepareValueType (self)) {
			//        return XGetter (p);
			// }
			// [DllImport ("libExpermimentalTests.dylib", EntryPoint = "_$s18ExpermimentalTests6OneIntV1XSivg")]
			// public unsafe static extern nint XGetter (byte* r);
```

Reply:
But it's not a byte* this is pass by value not pass by reference, so I can't really do that.

Reply:
Ah, I see, nvm.